File: subsystems/semantic-search/src/search-query.ts
/**
 * ==========================================================
 * SKOS
 * Smaily Knowledge Operating System
 * ==========================================================
 *
 * Subsystem : Semantic Search
 * Module    : Search Query
 *
 * Build     : BUILD-000189
 * Version   : 1.0.0
 * ==========================================================
 */

/**
 * Search execution mode.
 */
export enum SearchMode {

    Keyword = "keyword",

    Semantic = "semantic",

    Hybrid = "hybrid"

}

/**
 * Search scope.
 */
export enum SearchScope {

    All = "all",

    Knowledge = "knowledge",

    Graph = "graph",

    Vault = "vault",

    Metadata = "metadata"

}

/**
 * Search query.
 */
export interface SearchQuery {

    /**
     * Original user query.
     */
    text: string;

    /**
     * Search mode.
     */
    mode: SearchMode;

    /**
     * Search scope.
     */
    scope: SearchScope;

    /**
     * Maximum number of results.
     */
    limit: number;

    /**
     * Minimum accepted score.
     */
    minimumScore: number;

    /**
     * Optional filters.
     */
    filters?: Record<string, unknown>;

}

File: SKOS-Android/app/src/main/kotlin/com/smaily/skos/search/SearchQuery.kt
package com.smaily.skos.search

/**
 * ------------------------------------------------------------------
 * SKOS (Smaily Knowledge Operating System)
 * ------------------------------------------------------------------
 * SearchQuery
 *
 * Represents a search request.
 *
 * Version : 1.0.0
 * ------------------------------------------------------------------
 */
data class SearchQuery(

    /**
     * Search text.
     */
    val text: String,

    /**
     * Maximum number of results.
     */
    val limit: Int = 20,

    /**
     * Case-sensitive search.
     */
    val caseSensitive: Boolean = false,

    /**
     * Search in tags.
     */
    val includeTags: Boolean = true,

    /**
     * Search in metadata.
     */
    val includeMetadata: Boolean = false
)
